app/controllers/api/v1/messages_controller.rb

```diff
@@ -29,15 +29,17 @@ class MessagesController < Api::V1::ApiController
       param :offer_id, String, desc: "Return messages for offer id."
       param :item_id, String, desc: "Return messages for item id."
       param :order_id, String, desc: "Return messages for order id"
+      param :state, String, desc: "Message state (unread|read) to filter on"
       def index
         @messages = @messages.where(id: params[:ids].split(",")) if params[:ids].present?
-        @messages = @messages.where(offer_id: params[:offer_id]) if params[:offer_id].present?
-        @messages = @messages.where(order_id: params[:order_id]) if params[:order_id].present?
-        @messages = @messages.where(item_id: params[:item_id]) if params[:item_id].present?
+        @messages = @messages.where(offer_id: params[:offer_id].split(",")) if params[:offer_id].present?
+        @messages = @messages.where(order_id: params[:order_id].split(",")) if params[:order_id].present?
+        @messages = @messages.where(item_id: params[:item_id].split(",")) if params[:item_id].present?
+        @messages = @messages.with_user_read_state(User.current_user, params[:state]) if params[:state].present?
```

Comment:
@patrixr, could you think of any other name please ? 
I think `with_user_read_state` is ambiguous. With user read status is implying that the `read` messages should come.
But instead we are fetching messages here on the based of state which is coming in params.

Thanks

cc- @namrataukirde